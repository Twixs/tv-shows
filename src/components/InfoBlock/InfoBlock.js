import React from 'react';
import s from './InfoBlock.module.scss';

export const InfoBlock = ({ data }) => (
  <div className={s.InfoWrapper}>
    <img src={data?.image?.medium} alt={data.name} />
    <div className={s.Details}>
      <h2>{data.name}</h2>
      <p dangerouslySetInnerHTML={{ __html: data.summary }} />
    </div>
  </div>
);
