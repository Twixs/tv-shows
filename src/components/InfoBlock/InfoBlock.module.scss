.InfoWrapper {
  display: flex;

  @media screen and (max-width: 787px) {
    flex-wrap: wrap;
  }
}

.Details {
  margin-left: 20px;

  @media screen and (max-width: 787px) {
    margin-left: 0;
  }
}
