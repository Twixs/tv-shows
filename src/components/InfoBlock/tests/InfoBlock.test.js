import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { InfoBlock } from '../InfoBlock';

describe('InfoBlock component', () => {
  const data = {
    name: 'Test',
    summary: '<p>Testing InfoBlock</p>',
  };

  it('should match snapshot', () => {
    const tree = renderer.create(<InfoBlock data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { getByText } = render(<InfoBlock data={data} />);

    expect(getByText(/Testing InfoBlock/i)).toBeInTheDocument();
  });
});
