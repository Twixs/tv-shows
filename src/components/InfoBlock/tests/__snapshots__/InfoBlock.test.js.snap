// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBlock component should match snapshot 1`] = `
<div
  className="InfoWrapper"
>
  <img
    alt="Test"
  />
  <div
    className="Details"
  >
    <h2>
      Test
    </h2>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Testing InfoBlock</p>",
        }
      }
    />
  </div>
</div>
`;
